### ISSUE TITLE
Remove deprecation warnings from `rasa/utils/train_utils.py`

### ISSUE THREAD

**Issue by ancalita** [RasaHQ/rasa#9399]
**Context**:
In preparation for the 3.0 release, all deprecation warnings need to be removed.

These were belatedly identified in `rasa/utils/train_utils.py` in connection with DIET classifier:
- `update_deprecated_sparsity_to_density()`
```
    if WEIGHT_SPARSITY in config:
        rasa.shared.utils.io.raise_deprecation_warning(
            f"`{WEIGHT_SPARSITY}` is deprecated."
            f"Please update your configuration file to use"
            f"`{CONNECTION_DENSITY}` instead.",
            warn_until_version=NEXT_MAJOR_VERSION_FOR_DEPRECATIONS,
            docs=DOCS_URL_MIGRATION_GUIDE_WEIGHT_SPARSITY,
        )
        config[CONNECTION_DENSITY] = 1.0 - config[WEIGHT_SPARSITY]
```
- `update_deprecated_loss_type()`
```
    if config.get(LOSS_TYPE) == SOFTMAX:
        rasa.shared.utils.io.raise_deprecation_warning(
            f"`{LOSS_TYPE}={SOFTMAX}` is deprecated. "
            f"Please update your configuration file to use"
            f"`{LOSS_TYPE}={CROSS_ENTROPY}` instead.",
            warn_until_version=NEXT_MAJOR_VERSION_FOR_DEPRECATIONS,
        )
        config[LOSS_TYPE] = CROSS_ENTROPY
```
- `_replace_deprecated_option()`
```
    if isinstance(new_option, str):
        rasa.shared.utils.io.raise_deprecation_warning(
            f"Option '{old_option}' got renamed to '{new_option}'. "
            f"Please update your configuration file.",
            warn_until_version=warn_until_version,
        )
        return {new_option: config[old_option]}

    rasa.shared.utils.io.raise_deprecation_warning(
        f"Option '{old_option}' got renamed to "
        f"a dictionary '{new_option[0]}' with a key '{new_option[1]}'. "
        f"Please update your configuration file.",
        warn_until_version=warn_until_version,
    )
```
- `create_common_callbacks()`
```
        if tensorboard_log_level == "minibatch":
            tensorboard_log_level = "batch"
            rasa.shared.utils.io.raise_deprecation_warning(
                "You set 'tensorboard_log_level' to 'minibatch'. This value should not "
                "be used anymore. Please use 'batch' instead."
            )
```
- `_check_confidence_setting()`
```
    if component_config[MODEL_CONFIDENCE] == LINEAR_NORM:
        rasa.shared.utils.io.raise_deprecation_warning(
            f"{MODEL_CONFIDENCE} is set to `{LINEAR_NORM}`. We "
            f"introduced this option in Rasa Open Source 2.3.0, "
            f"but have identified multiple problems with it based "
            f"on user feedback. Therefore, `{MODEL_CONFIDENCE}={LINEAR_NORM}` "
            f"is now deprecated and will be removed in Rasa Open Source `3.0.0`."
            f"Please use `{MODEL_CONFIDENCE}={SOFTMAX}` instead."
        )
```
**Definition of Done**:
- [ ] the deprecation warnings are resolved.
- [ ] @samsucik assigned reviewer

**Comment by joejuzl** [comment]
There is also this is `diet_classifier.py`:

```
# TODO: This should be removed in 3.0 when softmax as
#  model confidence and normalization is completely deprecated.
message_sim = train_utils.normalize(
    message_sim, self.component_config[RANKING_LENGTH]
)
```

**Comment by jupyterjazz** [comment]
> There is also this is `diet_classifier.py`:
> 
> ```
> # TODO: This should be removed in 3.0 when softmax as
> #  model confidence and normalization is completely deprecated.
> message_sim = train_utils.normalize(
>     message_sim, self.component_config[RANKING_LENGTH]
> )
> ```

I have removed softmax as a loss type but I can't see a deprecation warning about softmax as model confidence. I'm just wondering if we should involve removing this code in this issue because softmax as model confidence will still exist after the changes I've made. Any ideas about that?

**Comment by samsucik** [comment]
@dakshvar22 maybe you can clarify the above-linked comment -- I think we're not removing softmax as model confidence/normalisation?

**Comment by dakshvar22** [comment]
Yes, the TODO in the code is outdated. `softmax` as a value for `model_confidence` should not be removed.
The issue with normalization (the code snippet above) should be handled separately because there are a few different ways to fix it in 3.0 and will need some feedback from CSE before we implement a solution.
I'd recommend to leave it out of this issue. I'll create another issue for it.
